### ISSUE TITLE
SSLException: No appropriate protocol

### ISSUE THREAD

**Issue by JiangXL** [scrapes/ILO2-Standalone-Remote-Console#12]
I'm trying to connect the ilo2 remote console at HP Proliant ML350 G6. After building, I got following error:
```bash
hf@spectre ~/b/ILO2-Standalone-Remote-Console (master) [1]> /usr/lib64/jvm/java-1.8.0-openjdk-1.8.0/bin/java -Djava.security.properties=java.security  -jar ILO2RemCon.jar -c config.properties
Couldn't find datastore, requesting Cookie
javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
        at sun.security.ssl.HandshakeContext.<init>(HandshakeContext.java:171)
        at sun.security.ssl.ClientHandshakeContext.<init>(ClientHandshakeContext.java:103)
        at sun.security.ssl.TransportContext.kickstart(TransportContext.java:220)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:433)
        at sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:559)
        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:197)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1572)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1500)
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:268)
        at Main.Stage1(Main.java:82)
        at Main.main(Main.java:301)
```
Any idea is welcome!

**Comment by fridtjof** [comment]
Are you running the latest ilo2 firmware? Should be 2.33, if I recall correctly. They implemented slightly more modern SSL with that, which is needed.

**Comment by JiangXL** [comment]
It is weird because my ilo2 is actually running the firmware 2.33.

**Comment by scrapes** [comment]
Can you post the Certificate information? Maybe it was issued with an older firmware.

**Comment by JiangXL** [comment]
Thank for your kind reply! I will post the certificate information later when I come back to laboratory.

I have flashed twice the 2.33 firmware today. Rather than firmware issue, I’m worrying the Java version and configuration. I try to build  and run with Java 8 and Java 11, but both of them show above error.

On Aug 10, 2022, at 19:10, Anton Scharnowski ***@***.***> wrote:

﻿

Can you post the Certificate information? Maybe it was issued with an older firmware.

**Comment by ppar** [comment]
FWIW -- I've been able to connect to the iLO2 web frontend using a Debian 9 VM. The openssl & co shipped with that OS still support the old ciphers.

Don't remember if this applies to the Java runtime too (and hence the remote console client).

My iLO2 firmware is probably a couple of versions behind too - too scared to update it.

**Comment by JiangXL** [comment]
Thanks everyone, here are how firefox shows:
```
Public Key Info
Algorithm: RSA
Key Size: 1024
Exponent: 65537
Modulus: xxxxxxxx
Miscellaneous
Signature Algorithm: MD5 with RSA Encryption
Version: NaN
Fringerprints
SHA-256: xxxxx
SHA-1: xxx
```

And web interface show: iLO 2 Firmware Version: | 2.33   03/20/2018

**Comment by scrapes** [comment]
You need to go into the cipher settings and set MD5 to SHA1 and reissue a new self signed certificate. (as far as I can remember) "Newer" systems don't support MD5 due to its being insecure.

**Comment by JiangXL** [comment]
Yes, I found that the existed certificate is pretty old which is issued at 2002. I try to issuse reissue self signed certificate, but I can not get the private key from ilo2 instead of a certificate request. Today, I generated a new private key, and issued certificate with this private key and the ilo2 certificate request. When I pasted the certificate to ilo2, it refused and showed that this certificate don't match private key in memory.

Thanks!

**Comment by JiangXL** [comment]
I finally the right certificate paired to  ilo2 certificate request by Let's encrypt. However, the SSLHandshakeException still occur on my openSUSE Tumbleweed which may be too "new". 

Why not try it on my old RPI 3b? It does work! The RPI 3b shipped with openjdk 1.8.0_312. Thanks again!

One more issue: The command in README `java -jar ILO2RemCon.jar -Djava.security.properties=java.security -c <Path to config.properties>` shows java argument error. The command `java -Djava.security.properties=java.security -jar ILO2RemCon.jar  -c <Path to config.properties>` works. If it happens all the time, the README will require a udpate.

**Comment by fridtjof** [comment]
I guess that might be specific to JDK 1.8, and I probably only tested that command on JDK 11. They might have relaxed the ordering requirements a bit. I'll adjust it anyway :)

**Comment by fridtjof** [comment]
fixed in eb0e50f5c88b57c455739c3d634a0b412400826f

**Comment by pcs3rd** [comment]
On the flipside, the most current variation of this command generates the following result: 
`Error: Could not find or load main class .security.properties=java.security`

Additionally, when executing in the original order, I get guidance on program usage:
Input: 
`java -jar ILO2RemCon.jar -c ..\config_template.properties -Djava.security.properties=../java.security`  
output: 
```
Usage:
- ILO2RemCon.jar <Hostname or IP> <Username> <Password>
- ILO2RemCon.jar -c <Path to config.properties>
```  
  
~~Contents of ..\config_template.properties:  
```
hostname = 172.168.x.x
username = uname
password = passwd
```~~
This actually was fixed by the filename, but the command execute path was still weird for me.

Environment details: 
```
PS C:\Users\uname> java -version
openjdk version "1.8.0_432"
OpenJDK Runtime Environment (Zulu 8.82.0.21-CA-win64) (build 1.8.0_432-b06)
OpenJDK 64-Bit Server VM (Zulu 8.82.0.21-CA-win64) (build 25.432-b06, mixed mode)
```
